connectors/salesforce/src/main/java/com/vmware/connectors/salesforce/SalesforceController.java

```diff
@@ -111,252 +117,281 @@ public SalesforceController(AsyncRestOperations rest, CardTextAccessor cardTextA
+            String baseUrl,
+            String soql
+    ) {
+        return fromHttpUrl(baseUrl)
+                .path(sfSearchAccountPath)
+                .queryParam("q", soql)
+                .build()
+                .toUri();
+    }
+
+    private URI makeUri(
+            String baseUrl,
+            String path
+    ) {
+        return fromHttpUrl(baseUrl)
+                .path(path)
+                .build()
+                .toUri();
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    // Cards request methods
+    ///////////////////////////////////////////////////////////////////
+
+    @PostMapping(
+            path = "/cards/requests",
+            consumes = APPLICATION_JSON_VALUE,
+            produces = APPLICATION_JSON_VALUE
+    )
     public Single<ResponseEntity<Cards>> getCards(
-            @RequestHeader(name = SALESFORCE_AUTH_HEADER) String sfAuth,
-            @RequestHeader(name = SALESFORCE_BASE_URL_HEADER) String sfBaseUrl,
-            @RequestHeader(name = ROUTING_PREFIX) String routingPrefix,
-            @Valid @RequestBody CardRequest cardRequest) {
+            @RequestHeader(SALESFORCE_AUTH_HEADER) String auth,
+            @RequestHeader(SALESFORCE_BASE_URL_HEADER) String baseUrl,
+            @RequestHeader(ROUTING_PREFIX) String routingPrefix,
+            @Valid @RequestBody CardRequest cardRequest
+    ) {
         // Sender email and user email are required, and sender email has to at least have a non-final @ in it
-        final String senderEmail = cardRequest.getTokenSingleValue("sender_email");
-        final String userEmail = cardRequest.getTokenSingleValue("user_email");
-        logger.debug("Sender email: {} and User email: {} for Salesforce server: {} ", senderEmail, userEmail, sfBaseUrl);
-        final String senderDomain = '@' + StringUtils.substringAfterLast(senderEmail, "@");
+        String sender = cardRequest.getTokenSingleValue("sender_email");
+        String user = cardRequest.getTokenSingleValue("user_email");
+        logger.debug("Sender email: {} and User email: {} for Salesforce server: {} ", sender, user, baseUrl);
+        String senderDomain = '@' + StringUtils.substringAfterLast(sender, "@");
         // TODO: implement a better system of validating domain names than "yup, it's not empty"
-        if (StringUtils.isBlank(senderDomain) || StringUtils.isBlank(userEmail)) {
-            logger.warn("Either sender email or user email is blank for url: {}", sfBaseUrl);
+        if (StringUtils.isBlank(senderDomain) || StringUtils.isBlank(user)) {
+            logger.warn("Either sender email or user email is blank for url: {}", baseUrl);
             return Single.just(new ResponseEntity<>(BAD_REQUEST));
         }
 
-        // TODO: concatenating strings into a SOQL query like this may provide an avenue for a SOQL-injection attack
-        String contactDetailsSoql = String.format(QUERY_FMT_CONTACT, senderEmail, userEmail);
```

Comment:
These are now being done serial instead of parallel.

The 3 potential REST calls are:
(1) Lookup single contact for combination of user+sender emails
(2) Lookup all contact opportunities for combination of user+sender emails
(3) Lookup all opportunities for the sender's account (because the sender might not exist)

In the previous code: First, it call call out to (1), (2), and (3) and then wait for (1) to return.  If (1) finds a contact, then we throw away the result of (3) and use (2) to create a card with the contact info.  If (1) does not find a contact, then we throw away the result of (2) and use (3) to create a card with an add-contact button.

In this PR, I've changed it to instead: Call out to (1) only then wait.  If (1) finds a contact, call into (2) and return the contact info card.  If (1) does not find a contact, then call into (3) and return the add-contact card.

(1) looks like it would probably be the fastest of the calls (pulling in less information than the others), so I'm choosing potentially a little more latency for better throughput (benefit from making 1/3 less calls).  There was no noticeable difference in my testing, so if there is a difference in latency, it's low milliseconds and/or under the normal variance of calls.

Reply:
600ms on the first call (after a restarted jvm, so there is presumably some priming for the connection pool)

sub-100ms on all followup calls (50-80ms or so)